samples/spring-security-basic-auth/src/main/java/sample/spring/xsuaa/security/TokenBrokerResolver.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * SPDX-FileCopyrightText: 2018-2023 SAP SE or an SAP affiliate company and Cloud Security Client Java contributors
+ * <p>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package sample.spring.xsuaa.security;
+
+import com.sap.cloud.security.xsuaa.tokenflows.PasswordTokenFlow;
+import com.sap.cloud.security.xsuaa.tokenflows.TokenFlowException;
+import com.sap.cloud.security.xsuaa.tokenflows.XsuaaTokenFlows;
+import jakarta.servlet.http.HttpServletRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.Cache;
+import org.springframework.http.HttpHeaders;
+import org.springframework.lang.Nullable;
+import org.springframework.security.oauth2.server.resource.web.BearerTokenResolver;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Optional;
+
+/**
+ * Resolves with an XSUAA user token from the configured service instance using the Basic HTTP credentials
+ * in the request header to execute a {@link com.sap.cloud.security.token.GrantType#PASSWORD} grant type flow.
+ * <p>
+ * The user crendentials are expected to be contained in the {@link HttpHeaders#AUTHORIZATION} header field as BASE64-encoded username/password combination.
+ * Optionally, a zone id may be specified via the 'X-Identity-Zone-Subdomain' header field.
+ * <p>
+ * For performance improvements, a pre-configured {@link Cache} may be supplied to limit the time between two token requests for the same principal.
+ */
+public class TokenBrokerResolver implements BearerTokenResolver {
+	private static final Logger logger = LoggerFactory.getLogger(TokenBrokerResolver.class);
```

Comment:
private static final usually is uppercase, `logger` should be consistent with other constants you defined e.g.  BASIC_AUTHENTICATION_SCHEME